packages/server/oauth/tokenHandler.ts

```diff
@@ -0,0 +1,158 @@
+import type {HttpRequest, HttpResponse} from 'uWebSockets.js'
+import {sign} from 'jsonwebtoken'
+import parseBody from '../parseBody'
+import getKysely from '../postgres/getKysely'
+import {Logger} from '../utils/Logger'
+
+const SERVER_SECRET = process.env.SERVER_SECRET!
+
+const tokenHandler = async (res: HttpResponse, _req: HttpRequest) => {
+  res.onAborted(() => {
+    res.aborted = true
+  })
+
+  try {
+    const rawBody = await parseBody({
+      res,
+      parser: (buffer: Buffer) => buffer.toString()
+    })
+
+    if (!rawBody) {
+      if (res.aborted) return
+      res.cork(() => {
+        res.writeStatus('400 Bad Request')
+        res.end(
+          JSON.stringify({error: 'invalid_request', error_description: 'Request body is missing'})
+        )
+      })
+      return
+    }
+
+    const params = new URLSearchParams(rawBody)
+    const body: any = {}
+    params.forEach((value, key) => {
+      body[key] = value
+    })
+
+    const redactedBody = {...body}
```

Comment:
-1 try to create objects in 1 call, it's a little cleaner on the compiler & has the added benefit of letting you know that this variable isn't used anywhere